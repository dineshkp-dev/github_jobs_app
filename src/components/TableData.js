import React from 'react';
import { Link } from 'react-router-dom';

function TableData(props) {
  const { type, title, location, id } = props.jobdetail;

  const link = '/jobdetails/' + id;

  return (
    <React.Fragment>
      <tr>
        <td>{type}</td>
        <td>
          <Link to={link}>{title}</Link>
        </td>
        <td>{location}</td>
      </tr>
    </React.Fragment>
  );
}

export default TableData;
