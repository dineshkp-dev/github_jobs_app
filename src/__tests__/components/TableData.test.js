import TableData from '../../components/TableData';
import { jobDetail } from '../utils/testUtils';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { BrowserRouter as Router } from 'react-router-dom';

describe('TableData snapshot test', () => {
  test('renders', () => {
    const { container } = render(
      <Router>
        <TableData jobdetail={jobDetail} />
      </Router>
    );
    expect(container).toMatchSnapshot();
  });
});
