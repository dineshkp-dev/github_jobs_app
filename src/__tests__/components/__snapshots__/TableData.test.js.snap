// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableData snapshot test renders 1`] = `
<div>
  <tr>
    <td>
      Full Time
    </td>
    <td>
      <a
        href="/jobdetails/testid"
      >
        Full Stack Engineer
      </a>
    </td>
    <td>
      New York City
    </td>
  </tr>
</div>
`;
